### PR TITLE
Adding shared_secret, adding cutorch build script, fully tested

### PR DESCRIPTION
**Pull request by soumith**
fully tested in jenkins at https://50.17.86.9:8443

### CODE REVIEW

**Comment by soumith** [comment]
Thanks a lot for this great repository :)

**Comment by hughperkins** [comment]
Cool! :-)

**Comment by hughperkins** [comment]
Some other things it might be good to consider at some point (just aesthetics really):
- get a dns record somewhere, eg in torch.ch?
- do the whole letsencrypt thing, to get rid of the self-signed?  (eg https://github.com/hughperkins/howto-jenkins-ssl/blob/master/letsencrypt.md )
- build name setter plugin https://wiki.jenkins-ci.org/display/JENKINS/Build+Name+Setter+Plugin , eg if you put `${GIT_BRANCH}_${GIT_REVISION,length=8}_#${BUILD_NUMBER}` if will give each build the name of the branch + commit hash + current build number

**Comment by hughperkins** [comment]
(pressed wrong button :-P )

File: jenkins/cutorch/build_nimbix.sh [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+git -c core.askpass=true fetch --tags https://github.com/torch/cutorch +refs/pull/*:refs/remotes/origin/pr/* --quiet
+git checkout $GIT_BRANCH
+time luarocks make rocks/cutorch-scm-1.rockspec 2>&1
+
+echo "Testing cutorch"
+time luajit -lcutorch -e "cutorch.test()" 2>&1
+
+echo "ALL CHECKS PASSED"
+
+# this is needed, i think because of a bug in nimbix-wrapper.py
```
Comment (soumith) [inline jenkins/cutorch/build_nimbix.sh:89]: just FYI, in case you use the wrapper for other projects.

**Comment by soumith** [comment]
just fyi, see line 89 of build_nimbix.sh
Also, this PR needs https://github.com/hughperkins/nimbix-admin/pull/1 to be merged

**Comment by hughperkins** [comment]
> Also, this PR needs hughperkins/nimbix-admin#1 to be merged 

Thanks!  Apparently I didnt have 'watch' on that repo.  Merged.

**Comment by hughperkins** [comment]
> just fyi, see line 89 of build_nimbix.sh

Ah.  Noted.  I guess a few hours of pain went into trying to track that down :-D

**Comment by soumith** [comment]
i didn't spend time tracking it down. just pushed a hack and it simply worked :)

**Comment by hughperkins** [comment]
:-)

**Pull request merged**